lib/block_view.go

```diff
@@ -4360,21 +4360,47 @@ func (bav *UtxoView) ConnectBlock(
 		}
 
 		if includeFeesInBlockReward {
-			// Add the fees from this txn to the total fees. If any overflow occurs
-			// mark the block as invalid and return a rule error. Note that block reward
-			// txns should count as having zero fees.
-			if totalFees > (math.MaxUint64 - currentFees) {
-				return nil, RuleErrorTxnOutputWithInvalidAmount
-			}
-			totalFees += currentFees
+			if txn.TxnMeta.GetTxnType() != TxnTypeAtomicTxnsWrapper {
+				// Compute the BMF given the current fees paid in the block.
+				_, utilityFee = computeBMF(currentFees)
```

Comment:
Am I crazy or does this call to computeBMF need to be gated behind a block height?

Reply:
Gated on line 4449.